RecoBTag/DeepFlavour/interface/sorting_modules.h

```diff
@@ -0,0 +1,187 @@
+    	sortValB=(rhs.sortValB);
+    	sortValC=(rhs.sortValC);
+    	t_=(rhs.t_);
+    	return *this;
+    }
+
+    const T& get()const{return t_;}
+
+    //hierarchical sort
+    static bool compareByABC(sortingClass a, sortingClass b){
+        if(std::isnormal(a.sortValA) && std::isnormal(b.sortValA) && a.sortValA!=b.sortValA){
+            return CompareA(a,b);
+        }
+        else if(!std::isnormal(a.sortValA) && std::isnormal(b.sortValA)){
+            return true;
+        }
+        else if(std::isnormal(a.sortValA) && !std::isnormal(b.sortValA)){
+            return false;
+        }
+        else{
+            if(std::isnormal(a.sortValB) && std::isnormal(b.sortValB) && a.sortValB!=b.sortValB){
+                return CompareB(a,b);
+            }
+            else if(!std::isnormal(a.sortValB) && std::isnormal(b.sortValB)){
+                return true;
+            }
+            else if(std::isnormal(a.sortValB) && !std::isnormal(b.sortValB)){
+                return false;
+            }
+            else{
+                if(std::isnormal(a.sortValC) && std::isnormal(b.sortValC) && a.sortValC!=b.sortValC){
+                    return CompareC(a,b);
+                }
+                else if(!std::isnormal(a.sortValC) && std::isnormal(b.sortValC)){
+                    return true;
+                }
+                else if(std::isnormal(a.sortValC) && !std::isnormal(b.sortValC)){
+                    return false;
+                }
+                else{
+                    return true;
+                }
+            }
+        }
+        return true; //never reached
+    }
+
+    static bool compareByABCInv(sortingClass a, sortingClass b){
+        return !compareByABC(a,b);
+    }
+
+ private:
+    float sortValA,sortValB,sortValC;
+    
+    static bool CompareA(sortingClass a, sortingClass b){
+        return a.sortValA<b.sortValA;
+    }
+    static bool CompareB(sortingClass a, sortingClass b){
+        return a.sortValB<b.sortValB;
+    }
+    static bool CompareC(sortingClass a, sortingClass b){
+        return a.sortValC<b.sortValC;
+    }
+    T t_;
+};
+
+
+
+std::vector<std::size_t> invertSortingVector(const std::vector<sortingClass<std::size_t> > & in);
+
+
+template<class T>
+bool compareDxy(T b, T a){
+    if(!a && b)return true;
+    if(!b && a)return false;
+    if(!a && !b)return false;
+    return a->dxy()<b->dxy();
+}
+
+template<class T>
+bool compareDxyDxyErr(T b, T a){
+    if(!a) return true;
+    if(!b) return false;
+    if(!a && !b)return false;
+
+    float aerr=a->dxyError();
+    float berr=b->dxyError();
+
+
+    float asig=a->dxy()/aerr;
+    float bsig=b->dxy()/berr;
+
+    if(!std::isnormal(asig) && std::isnormal(bsig))
+        return true;
+    else if(!std::isnormal(bsig) && std::isnormal(asig))
+        return false;
+    else if(!std::isnormal(bsig) && !std::isnormal(asig))
+        return false;
+
+    return asig<bsig;
+}
+
+
+template<class T>
+bool pfCCandSort(T b, T a){
```

Comment:
it looks like this is not used either

Reply:
checked with @jkieseler and not needed anymore, will remove

Reply:
removed with https://github.com/cms-sw/cmssw/pull/19893/commits/6ad48f2a3527cf49f3a753cf57bfe30380048472